### PR TITLE
Fix #8 #9

### LINKED ISSUE #8: Error thrown for projects with alpha version
The code throws an error while getting the release date of version alpha

### LINKED ISSUE #9: Json format error when writing version details file for the first time
Expected array but got dictionary

### PR DESCRIPTION
**Pull request by asa1997**
Fix #8
Fix #9

### CODE REVIEW

File: tests/test_metadata.py [hunk 1]
```diff
@@ -96,3 +96,20 @@ def test_issue_project_mismatch():
             raise AssertionError
         else:
             pass
+def test_version_alpha():
```
[the remark this answers is not in the document]
Comment (sudhirverma) [reply tests/test_metadata.py:176]: can you fix above sonatype suggestion.

File: tests/test_metadata.py [hunk 2]
```diff
@@ -96,3 +96,20 @@ def test_issue_project_mismatch():
...
+    osspoi = os.environ['OSSPOI_DIR']
+    result = runner.invoke(app, ["generate", "metadata"], input=str(id)+"\n"+name+"\n")
+    assert result.exit_code == 0
+    f = open(f"{osspoi}/version_details/{id}-{name}-Versiondetails.json")
+    data = json.load(f)
+    for i in range(len(data)):
+        if data[i]["version"] == version:
+            test_data = data[i]
+            break
+
```
[the remark this answers is not in the document]
Comment (sudhirverma) [reply tests/test_metadata.py]: can you fix whitespace issue

File: besecure_developer_toolkit/src/CreateVersionData.py [hunk 3]
```diff
@@ -32,9 +32,12 @@ def get_release_date(self, version, name):
         date = str(out).split(" ")[0]
         raw_date = date.split("'")[1]
         split_date = raw_date.split("-")
-        format_datetime = datetime.datetime(int(split_date[0]), int(split_date[1]), int(split_date[2]))
-        final_date = str(format_datetime.strftime("%d-%b-%Y"))    
-        return final_date
+        try:
+            format_datetime = datetime.datetime(int(split_date[0]), int(split_date[1]), int(split_date[2]))
```
[the remark this answers is not in the document]
Comment (sudhirverma) [reply besecure_developer_toolkit/src/CreateVersionData.py]: can you fix above mention issue.

File: besecure_developer_toolkit/src/CreateVersionData.py [hunk 4]
```diff
@@ -32,9 +32,12 @@ def get_release_date(self, version, name):
...
+            final_date = str(format_datetime.strftime("%d-%b-%Y"))    
+            return final_date
+        except Exception as e:
```
[the remark this answers is not in the document]
Comment (sudhirverma) [reply besecure_developer_toolkit/src/CreateVersionData.py]: can you skip pass from except try to show some log

File: besecure_developer_toolkit/src/CreateVersionData.py [hunk 5]
```diff
@@ -32,9 +32,12 @@ def get_release_date(self, version, name):
         date = str(out).split(" ")[0]
         raw_date = date.split("'")[1]
         split_date = raw_date.split("-")
-        format_datetime = datetime.datetime(int(split_date[0]), int(split_date[1]), int(split_date[2]))
-        final_date = str(format_datetime.strftime("%d-%b-%Y"))    
-        return final_date
+        try:
+            format_datetime = datetime.datetime(int(split_date[0]), int(split_date[1]), int(split_date[2]))
+            final_date = str(format_datetime.strftime("%d-%b-%Y"))
```
[the remark this answers is not in the document]
Comment (sudhirverma) [reply besecure_developer_toolkit/src/CreateVersionData.py]: same here can you fix above mention issue.

File: besecure_developer_toolkit/src/CreateVersionData.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sudhirverma) [reply besecure_developer_toolkit/src/CreateVersionData.py]: can you fix above mention issue for best practices.

File: README.md [hunk 6]
```diff
@@ -2,16 +2,16 @@
 
 bes-dev-kit is a cli tool for generating metadata and assessment report for [BeSLighthouse](https://github.com/Be-Secure/BeSLighthouse).
 
+# Pre-requisites
+
+1. Python 3.x
```
Comment (sudhirverma) [inline README.md]: I think this application support 3.10 or above? right??

File: besecure_developer_toolkit/src/CreateVersionData.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (sudhirverma) [reply besecure_developer_toolkit/src/CreateVersionData.py]: same here as above.

**Review by sudhirverma — CHANGES_REQUESTED** [review]
@asa1997 can you try to fix sonatype-lift comment

**Comment by sudhirverma** [comment]
@asa1997 can you resolve conflict

File: besecure_developer_toolkit/src/create_ossp_master.py [hunk 7]
```diff
@@ -17,73 +18,76 @@ def __init__(self, issue_id: int, name: str) -> None:
...
-        json_data = json.loads(urlopen(
-            f'https://api.github.com/repos/Be-Secure/Be-Secure/issues/{self.issue_id}').read())
-        issue_title = json_data["title"]
-        project_name = str(str(issue_title).split(":")[1]).replace(" ", "")
-        if project_name != self.name:
-            print(
-                f"[bold red]Alert! [yellow]Mismatch issue_id-project : [green] \
-                    Issue id {self.issue_id} does not match the project {self.name}")
-            sys.exit()
+        url = f'https://api.github.com/repos/Be-Secure/Be-Secure/issues/{self.issue_id}'
```
[the remark this answers is not in the document]
Comment (asa1997) [reply besecure_developer_toolkit/src/create_ossp_master.py:42]: @sonatype-lift ignore

File: besecure_developer_toolkit/src/create_ossp_master.py [hunk 8]
```diff
@@ -17,73 +18,76 @@ def __init__(self, issue_id: int, name: str) -> None:
...
+        with urllib.request.urlopen(url) as raw_data:
+            json_data = json.loads(raw_data.read())
+            issue_title = json_data["title"]
+            project_name = str(str(issue_title).split(":")[1]).replace(" ", "")
+            if project_name != self.name:
+                print(
+                    f"[bold red]Alert! [yellow]Mismatch issue_id-project : [green] \
```
Comment (sudhirverma) [inline besecure_developer_toolkit/src/create_ossp_master.py]: can you fix this string as I am getting lots of space in terminal

```
Enter OSSP name:owlviewer
Alert! Mismatch issue_id-project :                          Issue id 89 does not match the project owlviewer
```

**Review by sudhirverma — CHANGES_REQUESTED** [review]

File: besecure_developer_toolkit/src/create_ossp_master.py (hunk 8 above)
Comment (asa1997) [reply besecure_developer_toolkit/src/create_ossp_master.py]: @sudhirverma Done

**Review by sudhirverma — APPROVED** [review]

**Pull request merged**